### DISCUSSION TITLE
Where does `bundle install` put the updated version of bundler if we have `BUNDLE_PATH` set?

### DISCUSSION THREAD

**Discussion by mohamedhafez** [ruby/rubygems#7876] [Q&A] [answered]
On my development machine, if my Gemfile.lock specifies a verison of bundler different from the one installed on my system, doing a `bundle install` downloads that specific version of bundler, and I can see it later with `gem list bundler`

On my deployment machine however, where I use capistrano to deploy, after the deployment I don't see the new version of bundler with `gem list bundler`. I was thinking this was because I have `BUNDLE_PATH` set, but when I check at `$BUNDLE_PATH/jruby/3.1.0/gems/` I can see all my other gems in the correct subdirectory, but not the new version of bundler there either. (I also have  `BUNDLE_DEPLOYMENT: "true"` if that makes a difference)

Should I file a bug report, or am I missing something obvious, like is the new version of bundler installed somewhere else?

**Comment by deivid-rodriguez** [comment, marked as answer]
Hi! I did a quick test locally with the latest version of Bundler and the locked version of Bundler is correctly installed into `BUNDLE_PATH`. So, this seems related to capistrano?

**Reply by mohamedhafez** [reply]
What was happening is that capistrano-bundler's default behavior, unless its disabled by setting `:bundle_check_before_install` to `false`, is the following: it calls `bundle check`, and if it returns `0`, then it doesn't call `bundle install`. 

Come to think of it, perhaps that *is* an issue with bundler, that `bundle check` should return `1` if bundler itself needs to be updated? Just because if there ever was a security issue with bundler, we'd want to get updated.
